app/Observers/PaymentObserver.php

```diff
@@ -27,5 +29,22 @@ public function saved(Payment $payment): void
         ) {
             SendDuesPaymentReminder::dispatch($payment->payable->user);
         }
+
+        // this is pretty cursed but i don't have a better idea on guaranteeing exactly one receipt email from
+        // ~four save events
+        Cache::lock('send_payment_receipt_'.$payment->id, 5)->get(static function () use ($payment): void {
```

Comment:
Do you not need to catch the `LockTimeoutException` in the event the lock can't be acquired?

Reply:
Not sure what we would do with it. Just ignore it i guess?

Reply:
Ignoring it seems fine

Reply:
After further consideration i don't think this implementation will be a problem in practice, if it shows up in sentry I'll stick it in a try/catch